modules/dhcp_isc/dhcp_isc_main.rb

```diff
@@ -5,25 +5,15 @@ module Proxy::DHCP::ISC
   class Provider < ::Proxy::DHCP::Server
     include Proxy::Util
 
-    def initialize
-      super(Proxy::DhcpPlugin.settings.server)
-      @config_file = Proxy::DHCP::ISC::Plugin.settings.config
-      @leases_file = Proxy::DHCP::ISC::Plugin.settings.leases
+    def initialize(provider_settings = OpenStruct.new, main_settings = OpenStruct.new, service = nil)
```

Comment:
Why is only this class refactored in this way?  Perhaps leave it for now, and do them all together?